src/metabase/lib/measure.cljc

```diff
@@ -0,0 +1,116 @@
+                               :long-display-name display-name))]
+    (merge
+     (if-let [measure-metadata (resolve-measure query measure-id-or-name)]
+       (lib.metadata.calculation/display-info query stage-number measure-metadata)
+       {:effective-type    :type/Number
+        :display-name      (fallback-display-name)
+        :long-display-name (fallback-display-name)})
+     opts-to-merge)))
+
+(defmethod lib.metadata.calculation/column-name-method :measure
+  [query stage-number [_tag _opts measure-id-or-name]]
+  (or (when-let [measure-metadata (resolve-measure query measure-id-or-name)]
+        (lib.metadata.calculation/column-name query stage-number measure-metadata))
+      "measure"))
+
+(defmethod lib.metadata.calculation/metadata-method :measure
+  [query _stage-number [_ opts measure-id]]
+  (if-let [measure-meta (lib.metadata/measure query measure-id)]
+    {:lib/type     :metadata/column
+     :base-type    :type/Number
+     :name         (or (:name opts) (:name measure-meta) "measure")
+     :display-name (or (:display-name opts) (:name measure-meta) (fallback-display-name))}
+    {:lib/type     :metadata/column
+     :base-type    :type/Number
+     :name         "measure"
+     :display-name (fallback-display-name)}))
+
+(mu/defn available-measures :- [:maybe [:sequential {:min 1} ::lib.schema.metadata/measure]]
+  "Get a list of Measures that you may consider using as aggregation for a query. Only Measures that have the same
+  `table-id` as the `source-table` for this query will be suggested."
+  ([query]
+   (available-measures query -1))
+  ([query :- ::lib.schema/query
+    stage-number :- :int]
+   (when (zero? (lib.util/canonical-stage-index query stage-number))
+     (when-let [source-table-id (lib.util/source-table-id query)]
+       (let [measures (lib.metadata.protocols/measures (lib.metadata/->metadata-provider query) source-table-id)
+             measure-aggregations (into {}
+                                        (keep-indexed (fn [index aggregation-clause]
+                                                        (when (lib.util/clause-of-type? aggregation-clause :measure)
+                                                          [(get aggregation-clause 2) index])))
+                                        (lib.aggregation/aggregations query 0))]
+         (cond
+           (empty? measures)             nil
+           (empty? measure-aggregations) (vec measures)
+           :else                         (mapv (fn [measure-metadata]
+                                                 (let [aggregation-pos (-> measure-metadata :id measure-aggregations)]
+                                                   (cond-> measure-metadata
+                                                     aggregation-pos (assoc :aggregation-positions [aggregation-pos]))))
+                                               measures)))))))
+
+(def keep-me
+  "Var to keep ns loaded for side-effects."
+  nil)
```

Comment:
Uh, what? You don't need this to actually exist for the `keep-me` thing to work.